plugins/modules/device_configs_backup_workflow_manager.py

```diff
@@ -339,6 +397,74 @@ def get_device_ids_list(self):
         self.log("Backup of {0} devices out of {1} devices is possible".format(valid_device_count, original_valid_device_count), "INFO")
         return device_ids
 
+    def get_site_devices(self):
+        """
+        Fetches the list of device ids when site is passed along with other parameters
+        Args:
+            self: The instance of the class containing the 'config' attribute to be validated.
+        Returns:
+            device_in_site: The list of device ids based on the parameters passed by the user,
+                present in the passed site
+        Example:
+            Stored paramters like management ip address/ family can be used to fetch the device ids
+            list present at the given site
+        """
+
+        site = self.validated_config[0].get("site")
+        device_ids = self.get_device_ids_list()
+        dev_in_site = []
+        for dev_id in device_ids:
+            dev_response = self.dnac_apply['exec'](
+                family="devices",
+                function='get_device_detail',
+                params={"search_by": dev_id,
+                        "identifier": "uuid"},
+                op_modifies=True
+            )
+            self.log("Response collected from the API 'get_device_detail' {0}".format(dev_response), "DEBUG")
+            dev_response = dev_response.get("response")
```

Comment:
this is little confusing...

can we use like:
```
device_details_response = self.dnac_apply['exec'](
                family="devices",
                function='get_device_detail',
                params={"search_by": dev_id,
                        "identifier": "uuid"},
                op_modifies=True
            )

```
and then:

`device_details = device_details_response.get("response")`

Reply:
Changed